inc/class-upgrade.php

```diff
@@ -1628,4 +1630,26 @@ static function ( $duplicate ) use ( $object_type ) {
 		);
 		// phpcs:enable
 	}
+
+	/**
+	 * Sets the last_known_viewable_post_types option for the 19.11 upgrade routine.
+	 *
+	 * @return void
+	 */
+	public function set_last_known_public_post_types() {
+		$public_post_types = \array_keys( \YoastSEO()->helpers->post_type->get_public_post_types() );
```

Comment:
`get_public_post_types` only works well after the `init` hook. not all custom posttypes may have been registered yet. I think its best to register your function on 'shutdown' instead.
But that would mean that the watcher is reached during the first request before the setting is initialized.

Reply:
Looking at your other PR, you already do an early return after initializing the list when the list is not initialized yet.
https://github.com/Yoast/wordpress-seo/pull/19095/files#diff-e3aa0572be225b31c0f4b730b9386507e3dd5c188b37e7e55546cdca7e21955bR119-R123
I think that would completely remove the need for this upgrade routine

Reply:
You're right, I added that change yesterday and didn't think about this! Shall we use this PR just to update the version number? I can revert the two commits related to the unneeded initialization...

Reply:
Yes, sounds good 👍